src/Http/Http.Abstractions/src/Metadata/AcceptsMetadata.cs

```diff
@@ -0,0 +1,33 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.AspNetCore.Http.Metadata
+{
+    /// <summary>
+    /// Metadata that specifies the supported request content types.
+    /// </summary>
+    public sealed class AcceptsMetadata : IAcceptsMetadata
```

Comment:
Do we need this type to be public? Is it sufficient for the the interface to be public?

Reply:
yes it's needed in ```MinimalActionEndpointRouteBuilderExtensions``` to set the default json media type.

Reply:
Couldn't the type be internal? We don't need to define it in this assembly if it's only use is to satisfy the interface.

Reply:
It's currently being used in the following assemblies : ```Http.Extensions``` for setting accepts metadata in both the ```RequestDelegateFactory``` and  ``` Microsoft.AspNetCore.Routing```  for setting the default accepts ```application/json``` in the ```MinimalActionEndpointRouteBuilderExtensions``` . So I don't understand how making it internal would work, given it's needed in two separate assemblies.

Reply:
Since this is always read via the `IAcceptsMetadata` interface which is public, we can make it internal and shared source similar to [UrlDecoder.cs](https://github.com/dotnet/aspnetcore/blob/main/src/Http/Http.Abstractions/src/Microsoft.AspNetCore.Http.Abstractions.csproj#L27) and include it in both projects.

That said, I don't see where this is used in `Microsoft.AspNetCore.Routing` and in MVC we can probably use the updated `ConsumesAttribute` instead, so we might not need shared source either.